### PR TITLE
Adds checkbox to only show recently failed projects

### PR DESCRIPTION
**Pull request by altaudio**

### CODE REVIEW

File: redux/selectors/projects.js [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
+  return _.filter(projects, project => {
+    const lowerCaseProjectName = _.toLower(project.name)
+    return _.includes(lowerCaseProjectName, lowerCaseProjectQuery)
+  })
+}
+
+const filterRecentlyFailedProjects = projects => {
+  return _.filter(projects, project => {
+    return !_.chain(project.branches)
+      .filter(branch => branch.latestBuild.status === 'failed')
```
Comment (richardgill) [inline redux/selectors/projects.js]: `'failed'` is a constant I reckon

File: redux/selectors/projects.js [hunk 2]
```diff
@@ -1,21 +1,36 @@
...
+  state => state.filters,
+  (projectsState, filtersState) => {
+    // We would normally do this server side, but this allows us to give a nice selector example
+    const projectsFilteredByQuery = filterProjectsByQuery(projectsState.projects, filtersState.projectQuery)
+    const recentlyFailedProjects = filterRecentlyFailedProjects(projectsFilteredByQuery)
 
     return {
       ...projectsState,
-      projects: filteredProjects,
+      projects: filtersState.showFailed ? recentlyFailedProjects : projectsFilteredByQuery,
```
Comment (richardgill) [inline redux/selectors/projects.js]: does this mean you can use both filters simulataneously?

Comment (altaudio) [reply redux/selectors/projects.js]: Yeah, `filterRecentlyFailedProjects` takes `projectsFilteredByQuery` as it's arg

Comment (richardgill) [reply redux/selectors/projects.js]: how to code it is a seperate issue.

But I'd like to be able to check the box and search 'wash' and only see failures related to wash doctors.

Comment (altaudio) [reply redux/selectors/projects.js]: This does this already

File: redux/selectors/projects.js [hunk 3]
```diff
@@ -1,21 +1,38 @@
 import _ from 'lodash'
 import { createSelector } from 'reselect-change-memoize'
 
+const failedStatus = 'failed'
```
Comment (richardgill) [inline redux/selectors/projects.js]: a true constant `const` has uppercase `FAILED_STATUS`

**Pull request merged**